pkg/apis/core/types_cloudprofile.go

```diff
@@ -150,3 +153,24 @@ const (
 	// VolumeClassPremium is a constant for the premium volume class.
 	VolumeClassPremium string = "premium"
 )
+
+// VersionClassification is the logical state of a version according to https://github.com/gardener/gardener/blob/master/docs/proposals/05-versioning-policy.md
+type VersionClassification string
+
+var SupportedVersionClassifications = sets.NewString(string(ClassificationPreview), string(ClassificationSupported), string(ClassificationDeprecated))
+
+const (
+	// ClassificationPreview indicates that a version has recently been added and not promoted to "Supported" yet.
+	// ClassificationPreview versions will not be considered for automatic Kubernetes patch version updates.
+	ClassificationPreview VersionClassification = "preview"
+	// ClassificationSupported indicates that a patch version is the default version for the particular minor version.
+	// There is always exactly one supported Kubernetes patch version for every still maintained Kubernetes minor version.
+	// Supported versions are eligible for the automated Kubernetes patch version update for shoot clusters in Gardener.
+	ClassificationSupported VersionClassification = "supported"
+	// ClassificationDeprecated indicates that a patch version should not be used anymore, should be updated to a new version
+	// and will eventually expire.
+	// The time after which a deprecated version expires can be configured in the `kubernetesVersionManagement` and `machineImageVersionManagement` section of the CloudProfile Controller config.
+	// Every version that is neither in preview nor supported is deprecated.
+	// All patch versions of not supported minor versions are deprecated.
```

Comment:
I don't understand this sentence - what do you mean with "not supported minor versions"?

Reply:
Thanks for spotting. Removed it (that information should not be there and is part of the optional version management).